schema/query/projection_evaluator.go

```diff
@@ -0,0 +1,170 @@
+package query
+
+import (
+	"context"
+	"fmt"
+
+	"sync"
+
+	"github.com/rs/rest-layer/schema"
+)
+
+// Eval evaluate the projection on the given payload with the help of the
+// validator. The resolver is used to fetch payload of references outside of the
+// provided payload.
+func (p Projection) Eval(ctx context.Context, payload map[string]interface{}, validator schema.Validator, resolver ReferenceResolver) (map[string]interface{}, error) {
```

Comment:
If the interface is to return payloads, the interface name is probably better called "Items", or to use better query terminology a "Selection":

```go
type Selection interface{
        Items() []map[string]interface{}
        ResolveReference(ctx context.Context, path string, *Query) Selection
        Validator() shema.Validator
}
```

However it's designed, it would be pretty cool if the API allows a future possibility to provide custom RefrenceResolvers, e.g. to compose connections to _other_ REST APIs, which might come in handy for some micro-services.